packages/core/package.json

```diff
@@ -112,11 +112,15 @@
   },
   "peerDependencies": {
     "@types/react": "^16.8.4 || ^17.0.0 || ^18.0.0-0",
-    "react": "^16.8.4 || ^17.0.0 || ^18.0.0-0"
+    "react": "^16.8.4 || ^17.0.0 || ^18.0.0-0",
+    "react-native": ">=0.59"
```

Comment:
This is a breaking change for any package manager that doesn’t understand peerDependenciesMeta, and will force those react web consumers to install react-native. I very much doubt this is a desired approach.

I’d suggest a separate package for RN users, so non-RN users don’t have to interact with RN requirements.

Reply:
"force those react web consumers to install react-native" - isn't the only version of npm that does this npm 7+, which does understand optional?

Reply:
By force, i mean that `npm ls` will exit nonzero, which is true in every version of npm. Making someone’s dep graph invalid is a problem even if it’s not exposed by default on npm 3-6.

Reply:
Is there a reason it can't be compatible with RN without adding the peer dep?

Reply:
I guess the node module resolution algorithm will still work even without marked peerdeps.

Reply:
Only way .native will work is through metro so it's be really weird if RN wasn't found.